File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let anagramHash = {}
    for(let i = 0; i < strs.length; i++){
        sortedStr = strs[i].split('').sort().join('')
        if(!anagramHash[sortedStr]) anagramHash[sortedStr] = []
        anagramHash[sortedStr].push(strs[i])
    }
    
    let anagramArray = []
    for(anagram in anagramHash){
        anagramArray.push(anagramHash[anagram])
    }
    
    return anagramArray
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hash = {}
        for word in strs:
            sortedWord = ''.join(sorted(word))
            if sortedWord not in hash:
                hash[sortedWord] = [word]
            else:
                hash[sortedWord].append(word)
        final = []
        for value in hash.values():
            final.append(value)
        return final